packages/clerk-js/src/ui/utils/test/mockHelpers.ts

```diff
@@ -13,20 +13,33 @@ type DeepJestMocked<T> = T extends FunctionLike
       }
     : T;
 
-const mockProp = <T>(obj: T, k: keyof T) => {
+type MockMap<T = any> = {
+  [K in { [K in keyof T]: T[K] extends (...args: any[]) => any ? K : never }[keyof T]]?: jest.Mock<
+    // @ts-expect-error -- the typing seems to be working in practice
+    T[K]
+  >;
+};
```

Comment:
This type gives us a type-safe way to provide specific mock implementations to `mockMethodsOf`, see below for usage.